### ISSUE TITLE
Windows Headers Files

### ISSUE THREAD

**Issue by jlaura** [sminster/csm#5]
The windows makefile does not also include the headers (as the linux version does).

To include the headers it looks like the makefile needs to be updated as:

`all: $(HEADERS) $(LIBRARY)`

and 

```shell
install::
        $(RM) $(INSTDIR)/lib/$(LIBNAME).so
        $(CP) $(LIBRARY:%.dll=%.lib) $(INSTDIR)/lib
        $(CP) $(HEADERS) $(INSTDIR)/include/csm
```

Though I am not 100% sure?  If those are correct, I am happy to open a PR with the changes.  I do not have immediate access to a windows build env. to test.

**Comment by sminster** [comment]
there is some oddities with the Makefile.win32 file.  You cannot even use the all target there to get the library out.  I'm not sure why exactly.

**Comment by jlaura** [comment]
We are working on a win build on our [CSM implementation](https://github.com/USGS-Astrogeology/CSM-CameraModel/), so we can report on how that is progressing. We did swap to use [CMake](https://github.com/USGS-Astrogeology/csm/blob/master/CMakeLists.txt) as the build system though so it could be that this is a bit too divergent from the current system.

I am happy to PR in the CMake build with docs if the working group would find that a valuable contribution.

**Comment by sminster** [comment]
Using CMake would probably be better than my basic Makefiles :)
When building on Windows, do you use the `objects_to_def` script?

**Comment by jlaura** [comment]
No, which could be our linking problem. The linking problem is not on the CSM, but on our implementation and I have been attributing that to us needing to add some windows guards to the implementation.

What is / where is the `objects_to_def` script?  My windows knowledge is poor.

**Comment by sminster** [comment]
Oops, I thought it was under the `scripts` directory, but it looks like it isn't.  I'll have to look into adding it there.

**Comment by jessemapel** [comment]
@sminster Would it be possible to get the script for the def file? We can't get successful builds with the new cmake system without it. I'm currently trying to get the csm API library hosted on conda-forge so that everyone can just install it from there in the future (https://github.com/conda-forge/staged-recipes/pull/6912), but It cannot be integrated until it we can successfully build on windows.

**Comment by sminster** [comment]
@jessemapel I created PR #11 to deliver those scripts to the `scripts` directory

**Comment by jessemapel** [comment]
@sminster :+1: great thank you!

**Comment by jessemapel** [comment]
CMake has a convenient flag that does all of this for you, [WINDOWS_EXPORT_ALL_SYMBOLS](https://cmake.org/cmake/help/latest/prop_tgt/WINDOWS_EXPORT_ALL_SYMBOLS.html#prop_tgt:WINDOWS_EXPORT_ALL_SYMBOLS). You don't even need macros in the source code for it.
